### ISSUE TITLE
Certificate filepath too long

### ISSUE THREAD

**Issue by johnhultgren** [pact-foundation/pact-net#149]
The filepath to the EntrustnetSecureServerCertificationAuthority is really long.
This causes an error when trying to load a project containing this package if the path to the project contains just a few folders.

Is it possible to shorten the path to the certificate?

This is the path to the certificate:
\packages\PactNet.Windows.2.3.0\build\..\tools\pact-win32\lib\ruby\lib\ruby\gems\2.2.0\gems\bundler-1.9.9\lib\bundler\ssl_certs\EntrustnetSecureServerCertificationAuthority.pem".

**Comment by bethesque** [comment]
Unfortunately, the `lib\ruby\lib\ruby\gems\2.2.0\gems\bundler-1.9.9\lib\bundler\ssl_certs\EntrustnetSecureServerCertificationAuthority.pem` part of that path is dictated by the ruby package management tools itself. That's not something that can be changed. The only part that could be shortened is the `\packages\PactNet.Windows.2.3.0\build..\tools\pact-win32` part.

**Comment by bethesque** [comment]
I suspect that part is dictated by the .net package manager. Neil?

**Comment by neilcampbell** [comment]
Yeah the package manager conventions dictate the `\packages\PactNet.Windows.2.3.0\` part, so there isn't a huge amount of room for reducing the path size. My recommendation is move your code closer to the root of the drive. Is there any reason that this is not possible?

**Comment by bethesque** [comment]
Just wondering, @neilcampbell if it would be possible to make the location that the standalone was expected configurable, so that a separate standalone instance could be installed manually at a shorter path.

**Comment by johnhultgren** [comment]
@neilcampbell We have a common folder structure throughout the company containing 4 folders. But even with just 4 folders the total path is too long.

**Comment by clydeigc** [comment]
Any update on this? The issue with long paths is causing quite an inconvenience when building on Windows machines.

**Comment by neilcampbell** [comment]
@bethesque Nah I haven't done any work there.

@johnhultgren @clydeigc What version of Windows and VS? Are you able to create and link me to minimal sample project on Github that reproduces your issues?

**Comment by clydeigc** [comment]
@bethesque I understand the challenge in solving this problem and that you are blocked by another the author of the alternative packaging system. 

@neilcampbell I am using Windows 10 Enterprise, Visual Studio Enterprise 2017 Version 15.8.6, .Net Core 2.1. I will create a minimal sample project of Github to reproduce the issue, however it will occur in any project using Pact which has the bin folder in a folder with path at least 100 characters long. In my case it fails during build when it tries to copy the specification_provider.rb:
```
Error	MSB3021	Unable to copy file "C:\xxxxx\xxxxxxxxxxxxxxx\.nuget\packages\pactnet.windows\2.3.3\tools\pact-win32\lib\ruby\lib\ruby\gems\2.2.0\gems\bundler-1.9.9\lib\bundler\vendor\molinillo\lib\molinillo\modules\specification_provider.rb" to "bin\Debug\netcoreapp2.1\pact-win32\lib\ruby\lib\ruby\gems\2.2.0\gems\bundler-1.9.9\lib\bundler\vendor\molinillo\lib\molinillo\modules\specification_provider.rb". The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.	Test.SampleProject.ContractTests.Consumer	C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets	4544	
```

**Comment by clydeigc** [comment]
@neilcampbell Would a minimal sample project help you? It will basically be a new project with pact as a dependency. The issue is more related to where the project is stored on the system rather than what the project is.

**Comment by neilcampbell** [comment]
@clydeigc Yes please. That way I'm using the same sample as you.

**Comment by adamrodger** [comment]
Fixed in 4.0 with the move to Rust core library
